.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,33 @@
+{
+  "name": "OpenTelemetry Java Instrumentation Dev Environment",
+  "build": {
+    "dockerfile": "Dockerfile",
+    "context": ".."
+  },
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "vscjava.vscode-java-pack",
+        "vscjava.vscode-gradle",
+        "ms-vscode.gradle"
+      ]
+    }
+  },
+  "features": {
+    "ghcr.io/devcontainers/features/java:1": {
+      "version": "21"
+    },
+    "ghcr.io/devcontainers/features/node:1": {
+      "version": "16"
+    }
+  },
```

Comment:
what is this for?

and is the Java section relevant since using mcr.microsoft.com/devcontainers/java:1-21-bookworm base image?

if we can install Java this way, it seems nice, and then we can use any base image?